catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -1323,6 +1350,24 @@ private CswRecordCollection queryProductById(String id) throws CswException {
         return cswRecordCollection;
     }
 
+    private String getRange(String rangeHeader) throws UnsupportedQueryException {
+        String response = null;
+
+        if (rangeHeader != null) {
+            if (rangeHeader.startsWith(BYTES_EQUAL)) {
+                String tempString = rangeHeader.substring(BYTES_EQUAL.length());
+                if (tempString.contains("-")) {
+                    response = tempString.substring(0, tempString.lastIndexOf("-"));
```

Comment:
Is `123-456-888` a valid range? I can't see how it would be but if I passed in `bytes=123-456-888-` you would return that.